pkg/transport/generic_transport_controller.go

```diff
@@ -304,13 +305,11 @@ func isObjectBeingDeleted(object metav1.Object) bool {
 }
 
 func (c *genericTransportController) deleteWrappedObjectsAndFinalizer(ctx context.Context, placementDecision *v1alpha1.PlacementDecision) error {
-	for _, destination := range placementDecision.Spec.Destinations { // TODO need to revisit the destination struct and see how to use it properly
-		if err := c.transportClient.Resource(c.wrappedObjectGVR).Namespace(destination.ClusterId).Delete(ctx, fmt.Sprintf("%s-%s", placementDecision.GetName(), c.wdsName),
-			metav1.DeleteOptions{}); err != nil { // wrapped object name is in the format (PlacementDecision.GetName()-WdsName). see updateWrappedObject func for explanation.
-			if !errors.IsNotFound(err) { // if object is already not there, we do not report an error cause desired state was achieved.
-				return fmt.Errorf("failed to delete wrapped object '%s' in destination WEC with namespace '%s' - %w", fmt.Sprintf("%s-%s", placementDecision.GetName(),
-					c.wdsName), destination.ClusterId, err)
-			}
+	for _, destination := range placementDecision.Spec.Destinations {
+		if err := c.deleteWrappedObject(ctx, destination.ClusterId, fmt.Sprintf("%s-%s", placementDecision.GetName(), c.wdsName)); err != nil {
+			// wrapped object name is in the format (PlacementDecision.GetName()-WdsName). see updateWrappedObject func for explanation.
+			return fmt.Errorf("failed to delete wrapped object '%s' in destination WEC mailbox namespace '%s' - %w", fmt.Sprintf("%s-%s", placementDecision.GetName(),
```

Comment:
FYI, this is repeating information in the wrapped error.

Reply:
fixed